### ISSUE TITLE
I HAVE A GOOD IDEA

### ISSUE THREAD

**Issue by mollthecoder** [glixzzy/blooket-wrapper#16]
Read carefully.

First of all, using blooket's exposed internal API is great and all, but doesn't allow you to do much. What if there was a client-side API that can control the website (Using APIs like ``<HTML ELEMENT>.click()`` and ``document.dispatchEvent(<EVENT>)``)  then developers can truly have all the power a player normally would. It wouldn't be as fast and a lot less easy to set up, but it would be a lot more powerful. 

Then combine that with the internal API to enhance the speed and capabilities further. Then it would be possible use the ``jsdom`` npm module to port the API to Node.js. By loading Blooket in ``jsdom`` we can execute JavaScript on behalf of blooket.

In conclusion, I envision the ability to do stuff like bots that can automatically host blooket games, for automated events, and things like virtual opponents for those who wish to play multiplayer blookets but without needing to assemble a group. But to do it, I need your help. What do you say?

**Comment by donaldli2020** [comment]
sounds cool

**Comment by gimmy74** [comment]
> Read carefully.
> 
> First of all, using blooket's exposed internal API is great and all, but doesn't allow you to do much. What if there was a client-side API that can control the website (Using APIs like `<HTML ELEMENT>.click()` and `document.dispatchEvent(<EVENT>)`) then developers can truly have all the power a player normally would. It wouldn't be as fast and a lot less easy to set up, but it would be a lot more powerful.
> 
> Then combine that with the internal API to enhance the speed and capabilities further. Then it would be possible use the `jsdom` npm module to port the API to Node.js. By loading Blooket in `jsdom` we can execute JavaScript on behalf of blooket.
> 
> In conclusion, I envision the ability to do stuff like bots that can automatically host blooket games, for automated events, and things like virtual opponents for those who wish to play multiplayer blookets but without needing to assemble a group. But to do it, I need your help. What do you say?

This is most likely not possible

**Comment by glixzzy** [comment]
no

**Comment by Poly-cryptic** [comment]
This is theoretically possible, much simpler too, with Puppeteer. However this is obviously not practical at all.
